zerver/openapi/openapi.py

```diff
@@ -45,6 +88,15 @@ def reload(self) -> None:
             yaml_parser = YamoleParser(f)
 
         self.data = yaml_parser.data
+        self.regex_dict = {}
+        for keys in self.data['paths']:
+            if '{' not in keys:
+                continue
+            regex_key = '^' + keys + '$'
+            regex_key = re.sub(r'{[^}]*}', r'[^\/]*', regex_key)
+            regex_key = regex_key.replace(r'/', r'\/')
+            regex_key = fr'{regex_key}'
+            self.regex_dict[regex_key] = keys
```

Comment:
Can you extract the OpenAPI Pattern => regex conversion to a function, with documentation explaining the algorithm and a few unit tests?

Reply:
Made the code more modular and added documentation. Going to push another commit to add tests.

Reply:
Added tests.